codec/src/test/java/io/netty/handler/codec/compression/JZlibTest.java

```diff
@@ -26,4 +28,10 @@ protected ZlibEncoder createEncoder(ZlibWrapper wrapper) {
     protected ZlibDecoder createDecoder(ZlibWrapper wrapper) {
         return new JZlibDecoder(wrapper);
     }
+
+    @Test(expected = DecompressionException.class)
+    @Override
+    public void testZLIB_OR_NONE3() throws Exception {
```

Comment:
This is a mistake. I simply copied the behavior from `JdkZlibTest` without realizing that JZlib library handles gzip decompression when the `ZlibWrapper` type is `ZLIB_OR_NONE` (which is mapped to `JZlib.W_ANY` and this handles gzip).

However when I tried to fix this, I realized that `JdkZlibDecoder` does not handle gzip decompression when the Wrapper type is `ZLIB_OR_NONE`. Only `JZlibDecoder` does this as it finally maps `ZLIB_OR_NONE` to JZlib's `JZlib.W_ANY` which handles gzip. Is it okay if `ZLIB_OR_NONE` handles only zlib or none and not gzip? In this case, we can remove the test case `testZLIB_OR_NONE3()` altogether. 

If `JZlibDecoder` should handle gzip for `ZLIB_OR_NONE`, then then I can detect the compression format in decode (using a new method like `ZlibUtil.looksLikeGzip()`) and then set the JZlib's wrapper type as `W_ANY`.

Please let me know your thoughts and I will modify the PR.